base/array.jl

```diff
@@ -324,9 +324,13 @@ copyto!(dest::Array{T}, src::Array{T}) where {T} = copyto!(dest, 1, src, 1, leng
 # N.B: The generic definition in multidimensional.jl covers, this, this is just here
 # for bootstrapping purposes.
 function fill!(dest::Array{T}, x) where T
-    xT = x isa T ? x : convert(T, x)::T
+    @inline
+    x = x isa T ? x : convert(T, x)::T
```

Comment:
The isa branch is probably now inserted by inference dispatch?

Reply:
Not sure why since the dispatch below is with a different name, but do you think something should be done about it?

Reply:
Or do you mean that this guard is obsolete now because of other improvements in inference?

Reply:
Ah, I didn't notice that was a different function then. The guard gives inference the opportunity to union-split it (or at least the option to ignore the convert) when encountering abstract types

Reply:
`convert(ConcreteType, x::AbstractType)` may generally have more than 4 methods, in which case the `isa` branch ins't generated by the inliner, e.g.
```julia
julia> methods(convert, (Type{Int},Any))
# 7 methods for generic function "convert" from Base:
 [1] convert(::Type{T}, x::Ptr) where T<:Integer
     @ pointer.jl:23
 [2] convert(::Type{T}, index::CartesianIndex{1}) where T<:Number
     @ Base.IteratorsMD multidimensional.jl:137
 [3] convert(::Type{T}, x::AbstractChar) where T<:Number
     @ char.jl:192
 [4] convert(::Type{T}, x::Base.TwicePrecision) where T<:Number
     @ twiceprecision.jl:271
 [5] convert(::Type{T}, x::T) where T<:Number
     @ number.jl:6
 [6] convert(::Type{T}, x::Number) where T<:Number
     @ number.jl:7
 [7] convert(::Type{T}, x::T) where T
     @ Base.jl:129
```